### PR TITLE
38679 Custom cookie in check-in experience for low auth token

### PR DESCRIPTION
**Pull request by kanchanasuriya**
## Description of change

Currently, minimal authentication in session object gets reset with the user signing out of full authentication in the system.   This PR moves minimal auth token to custom cookie and avoids minimal authentication getting interrupted with full authentication. 

## Original issue(s)
department-of-veterans-affairs/va.gov-team#38679

## Things to know about this PR

- [x] rubocop
- [x] rspec

### CODE REVIEW

**Review by thilton-oddball — DISMISSED** [review]

File: modules/check_in/app/controllers/check_in/application_controller.rb [hunk 1]
```diff
@@ -25,6 +25,22 @@ def additional_logging?
       Flipper.enabled?('check_in_experience_logging_enabled')
     end
 
+    def minimal_auth_token
```
Comment (randomsync) [inline modules/check_in/app/controllers/check_in/application_controller.rb:28]: we seem to be using the terms minimal auth and low auth interchangeably. should we standardize on one of them, maybe low auth?

**Review by randomsync — COMMENTED** [review]
@kanchanasuriya are the test scenarios for this change listed somewhere?

File: modules/check_in/app/controllers/check_in/application_controller.rb (hunk 1 above)
Comment (kanchanasuriya) [reply modules/check_in/app/controllers/check_in/application_controller.rb:28]: Maybe, I will make it minimal auth for the feature flag.  If we change everything to low auth, then the cookies object having `min_session` will be out of sync.  

At the same time, not inlined in using low_auth_session in cookie object as it is visible to the client. Please let me know.

Comment (kanchanasuriya) [reply modules/check_in/app/controllers/check_in/application_controller.rb]: As discussed offline, modified the method name and cookie name in the latest commit

**Review by thilton-oddball — APPROVED** [review]

**Review by randomsync — APPROVED** [review]
lgtm

**Pull request merged**